types/index.d.ts

```diff
@@ -0,0 +1,318 @@
+interface AuthOptions {
+  clientId: string;
+  persistentCache: boolean;
+  redirectUri: string;
+}
+
+interface AzureAuthOptions {
+  /** @param {String} clientId your AzureAuth client identifier */
+  clientId: string;
+  /** @param {boolean} [persistentCache] should store token cache between the app starts; defaults to true */
+  persistentCache?: string;
+  /** @param {String} [authorityUrl] optional Azure authority if you want to replace default v2 endpoint (`https://login.microsoftonline.com/${tenant}/oauth2/v2.0/`) */
+  authorityUrl?: string;
+  /** @param {String} [tenant] uses given tenant in the default authority URL; default is `common` */
+  tenant?: string;
+  /** @param {String} [redirectUri] - uses given redirect URI instead of default one:
+   *
+   *  iOS: {YOUR_BUNDLE_IDENTIFIER}://${YOUR_BUNDLE_IDENTIFIER}/ios/callback
+   *
+   *  Android: {YOUR_APP_PACKAGE_NAME}://{YOUR_APP_PACKAGE_NAME}/android/callback
+   */
+  redirectUri?: string;
+}
+
+interface ClientOptions {
+  token: string;
+  authorityUrl?: string;
+  baseUrl?: string;
+}
+
+interface TokenResponse {
+  accessToken: string;
+  tokenType: string;
+  expiresIn: number;
+  scope: string;
+  refreshToken: string;
+  idToken: string;
+}
+
+/**
+ * Helper to perform HTTP requests
+ * Blob (binary) content types are not supported
+ *
+ * Class variables:
+ * - baseUrl | authorityUrl: base URL the request path is added to
+ * - token: Auth token if the request needs authorization
+ *
+ * One of Url options must be provided
+ *
+ * @param options
+ * @param options.authorityUrl
+ * @param options.baseUrl
+ * @param options.token
+ */
+declare class Client {
+  constructor(options: ClientOptions);
+  baseUrl: string;
+  bearer: string;
+  get(path: string, query: string): Promise<void>;
+  patch(path: string, body: any): Promise<void>;
+  post(path: string, body: any): Promise<void>;
+  /**
+   * Helper function to send HTTP requests
+   *
+   * @param {String} method
+   * @param {String} url
+   * @param {Object} [body] - request body
+   */
+  request(method: string, url: string, body?: any): Promise<void>;
```

Comment:
The type argument for the `Promise` is most definitely [not void](https://github.com/vmurin/react-native-azure-auth/blob/v1.5.1/src/networking/index.js#L91).
Considering the following interface def:

interface HttpResponse {
	status:string,
	ok: boolean,
    headers: Headers,
    blob?: Blob,
    json?: object,
    text?: string
}

```suggestion
  request(method: string, url: string, body?: any): Promise<HttpResponse>;
```

Reply:
Thanks! Very good suggestion!
@jotatoledo could you please add the interface as "_suggested change_" so I can take it over to PR
But I just would call it like `ClientResponse` as the `HttpResponse` is too common name and can interfere with something else and irritate ;).